recipe/meta.yaml

```
@@ -1,22 +1,21 @@
-{% set version = "3.0.3" %}
+{% set version = "3.1.1" %}
 
 package:
   name: libjpeg-turbo
   version: {{ version }}
 
 source:
   url: https://github.com/libjpeg-turbo/libjpeg-turbo/archive/{{ version }}.tar.gz
-  sha256: a649205a90e39a548863a3614a9576a3fb4465f8e8e66d54999f127957c25b21
+  sha256: 304165ae11e64ab752e9cfc07c37bfdc87abd0bfe4bc699e59f34036d9c84f72
 
 build:
   number: 0
-  # We don't have yasm for s390x; libjpeg-turbo is needed for torchvision, which we won't build on s390x anyway
-  skip: True # [s390x]
   run_exports:
     - {{ pin_subpackage('libjpeg-turbo', max_pin='x') }}
 
 requirements:
   build:
+    - {{ stdlib('c') }}
     - {{ compiler('c') }}
     - cmake
```

```suggestion
    - cmake >=2.8.12
```

Don't we need to specify _cmake_ version range based on _cmake_minimum_required_?

https://github.com/libjpeg-turbo/libjpeg-turbo/blob/3.1.1/CMakeLists.txt#L1

---

Technically, yes but what this is really doing in the upstream is setting a minimum CMake version and a maximum version of the rules and policies that CMake operates under. By default, CMake will use the oldest set of compatible policies listed in that range so would _pretend_ that it is v3.28 even if it was something much newer. In our case, since we are installing packages from our main channel, it is not possible to accidentally end up with a version of CMake old enough to cause build issues since our oldest available is v3.9.1. The upper bound can also be omitted here since our newer CMake will simply operate with the policies that applied to v3.28.

---

Understandable, but there is also note in the upstream: https://github.com/libjpeg-turbo/libjpeg-turbo/blob/3.1.1/BUILDING.md?plain=1#L11

I understand that we're not having such old version, but I have seen other requirements where we limit their versions from far below the oldest we have in main channel (e.g. `setuptools >=41` where the oldest available is 49.1.3). Doesn't it cause inconsistency here?

---

Consistency is somewhat difficult to come by with this many people working on these recipes. There are quite a few recipes where the upstream has a pin on `setuptools >=69` which we may or may not add to our recipe since we know the version will be satisfied and that it's only a build dependency and has no real impact on the package produced so long as the build succeeds.